libkatana_python_native/include/katana/python/NumbaSupport.h

```diff
@@ -0,0 +1,146 @@
+#ifndef KATANA_LIBKATANAPYTHONNATIVE_KATANA_PYTHON_NUMBASUPPORT_H_
+#define KATANA_LIBKATANAPYTHONNATIVE_KATANA_PYTHON_NUMBASUPPORT_H_
+
+#include <type_traits>
+
+#include <katana/python/TypeTraits.h>
+#include <pybind11/pybind11.h>
+
+namespace katana {
+
+// TODO(amp): Use std::remove_cvref_t when/if we are only supporting C++20.
+template <typename T>
+using remove_cvref_t = std::remove_cv_t<std::remove_reference_t<T>>;
+
+/// A wrapper around a member function pointer which (unlike std::mem_fn) allows
+/// calls to the function via a static function pointer allowing it to be
+/// called from Numba generated code.
+///
+/// @tparam Cls The class containing the method.
+/// @tparam Return The return value of the method.
+/// @tparam Args The argument types to the method.
+template <typename Cls, typename Return, typename... Args>
+struct MemberFunction {
+  Return (Cls::*member_func)(Args...);
+
+  static remove_cvref_t<Return> call(
+      MemberFunction<Cls, Return, Args...>* func, Cls* self,
+      remove_cvref_t<Args>... args) {
+    return (self->*(func->member_func))(args...);
+  }
+};
+
+template <typename Cls, typename Return, typename... Args>
+struct ConstMemberFunction {
+  Return (Cls::*member_func)(Args...) const;
+
+  static remove_cvref_t<Return> call(
+      ConstMemberFunction<Cls, Return, Args...>* func, const Cls* self,
+      remove_cvref_t<Args>... args) {
+    return (self->*(func->member_func))(args...);
+  }
+};
+
+template <typename... Args>
+class DefWithNumbaImpl {
```

Comment:
Where and how is Numba being used? IIRC, it's used for compiling python code, correct?

Reply:
Numba is what we use to compile user code that will run inside galois/katana parallel loops. So it's used in the `@do_all_operator` and `@for_each_operator` function decorators (indirectly, but that's where it's triggered).